azurerm/internal/services/storage/resource_arm_storage_account_customer_managed_key.go

```diff
@@ -0,0 +1,249 @@
+	storageClient := meta.(*clients.Client).Storage.AccountsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	storageAccountId := d.Get("storage_account_id").(string)
+	keyVaultId := d.Get("key_vault_id").(string)
+
+	d.Set("storage_account_id", storageAccountId)
+	d.Set("key_vault_id", keyVaultId)
+
+	name, resGroup, err := azure.ParseNameAndResourceGroupFromResourceId(storageAccountId, "storageAccounts")
+	if err != nil {
+		return err
+	}
+
+	resp, err := storageClient.GetProperties(ctx, resGroup.(string), name.(string), "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading the state of AzureRM Storage Account %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if props := resp.AccountProperties; props != nil {
+		if encryption := props.Encryption; encryption != nil {
+			if keyVaultProperties := encryption.KeyVaultProperties; keyVaultProperties != nil {
+				d.Set("key_name", *keyVaultProperties.KeyName)
+				d.Set("key_version", *keyVaultProperties.KeyVersion)
+				d.Set("key_vault_uri", *keyVaultProperties.KeyVaultURI)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmStorageAccountCustomerManagedKeyDelete(d *schema.ResourceData, meta interface{}) error {
+	storageClient := meta.(*clients.Client).Storage.AccountsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	storageAccountId := d.Get("storage_account_id").(string)
+	encryptionServices := true
+
+	storageAccountName, resourceGroupName, err := azure.ParseNameAndResourceGroupFromResourceId(storageAccountId, "storageAccounts")
+	if err != nil {
+		return err
+	}
+
+	// Since this isn't a real object, just modifying an existing object
+	// "Delete" doesn't really make sense it should really be a "Revert to Default"
+	// So instead of the Delete func actually deleting the Storage Account I am
+	// making it reset the Storage Account to it's default state
+	props := storage.AccountUpdateParameters{
+		AccountPropertiesUpdateParameters: &storage.AccountPropertiesUpdateParameters{
+			Encryption: &storage.Encryption{
+				Services: &storage.EncryptionServices{
+					Blob: &storage.EncryptionService{
+						Enabled: utils.Bool(encryptionServices),
+					},
+					File: &storage.EncryptionService{
+						Enabled: utils.Bool(encryptionServices),
+					},
+				},
+				KeySource: storage.MicrosoftStorage,
+			},
+		},
+	}
+
+	_, err = storageClient.Update(ctx, resourceGroupName.(string), storageAccountName.(string), props)
+	if err != nil {
+		return fmt.Errorf("Error deleting AzureRM Storage Account %q (Resource Group %q) %q: %+v", storageAccountName.(string), resourceGroupName.(string), err)
+	}
+
+	return nil
+}
+
+func resourceArmStorageAccountCustomerManagedKeyImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
this is unused - but we can look this information up from the storage account in the read so this shouldn't be needed